xml-parser/src/main/java/org/uniprot/core/xml/uniprot/comment/SubcellLocationNameMapImpl.java

```diff
@@ -50,7 +50,7 @@ public String getLocationName(String name) {
     private void loadSubcellularLocationMap(String filename) {
         if (Utils.notNullNotEmpty(filename)) {
             List<SubcellularLocationEntry> entries =
-                    new SubcellularLocationFileReader().parse(filename);
+                    SubcellularLocationCache.INSTANCE.get(filename);
             subcellularLocationMap =
                     entries.stream()
                             .collect(
```

Comment:
when the `toMap` is called, do you know if it is modifiable or not? I notice in one constructor, the map is made unmodifiable. Therefore, I think we should also make sure this generated map is also unmodifiable.